ocaml/xapi/repository.ml

```diff
@@ -544,6 +563,28 @@ let apply_immediate_guidances ~__context ~host ~guidances =
       (ExnHelper.string_of_exn e) ;
     raise Api_errors.(Server_error (apply_guidance_failed, [ref]))
 
+let apply_pending_guidances ~__context ~host ~guidances =
+  try
+    let open Guidance in
+    guidances
+    |> List.iter (function
+         | RebootHost ->
+             Db.Host.add_pending_guidances ~__context ~self:host
+               ~value:`RebootHost
+         | RestartToolstack ->
+             Db.Host.add_pending_guidances ~__context ~self:host
+               ~value:`RestartToolstack
+         | RestartDeviceModel ->
+             set_pending_restart_device_models ~__context ~host
+         | g ->
+             warn "Unsupported pending guidance %s, ignore it."
+               (Guidance.to_string g))
+  with e ->
+    let ref = Ref.string_of host in
```

Comment:
`ref` is a bad name - it refers to a host; Something like `host'` would be better.